scripts/ci-check.sh

```
@@ -164,6 +261,52 @@ render_command() {
+    return 1
+  fi
+
+  local stamp_file="${target_dir}/${CI_CHECK_TOOLCHAIN_STAMP}"
+  local current_fingerprint=""
+  if [[ -f "${stamp_file}" ]]; then
+    current_fingerprint="$(cat "${stamp_file}" 2>/dev/null || true)"
+  fi
+
+  if [[ -d "${target_dir}" && "${current_fingerprint}" != "${CI_CHECK_TOOLCHAIN_FINGERPRINT}" ]]; then
+    rm -rf -- "${target_dir}"
+    echo "info: ${target_dir#${REPO_ROOT}/} を削除しました (rustc/toolchain が変わったため)" >&2
```

Shellcheck (0.11.0)

[info] Expansions inside ${..} need to be quoted separately, otherwise they match as patterns.

(SC2295)